### ISSUE TITLE
Trying to insert guest customer with null CountryId throws exception

### ISSUE THREAD

**Issue by sencagri** [nopSolutions/nopCommerce#6289]
nopCommerce branch: 4.5 bug-fixes

Steps to reproduce the problem: In some state I don't know how to reproduce this. I got this error message which is super understandable. Country Id is null so we cannot add this to the customer table. Is this a bug?

```
Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, bool breakConnection, Action<Action> wrapCloseInAction)
Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, bool callerHasConnectionLock, bool asyncClose)
Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, out bool dataReady)
Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, string resetOptionsString, bool isInternal, bool forDescribeParameterEncryption, bool shouldCacheForAlwaysEncrypted)
Microsoft.Data.SqlClient.SqlCommand.CompleteAsyncExecuteReader(bool isInternal, bool forDescribeParameterEncryption)
Microsoft.Data.SqlClient.SqlCommand.InternalEndExecuteReader(IAsyncResult asyncResult, bool isInternal, string endMethod)
Microsoft.Data.SqlClient.SqlCommand.EndExecuteReaderInternal(IAsyncResult asyncResult)
Microsoft.Data.SqlClient.SqlCommand.EndExecuteReaderAsync(IAsyncResult asyncResult)
System.Threading.Tasks.TaskFactory<TResult>.FromAsyncCoreLogic(IAsyncResult iar, Func<IAsyncResult, TResult> endFunction, Action<IAsyncResult> endAction, Task<TResult> promise, bool requiresSynchronization)
LinqToDB.Data.DataConnection.ExecuteScalarAsync(CancellationToken cancellationToken)
LinqToDB.Data.DataConnection+QueryRunner.ExecuteScalarAsync(CancellationToken cancellationToken)
LinqToDB.Linq.QueryRunner.ScalarQueryAsync(Query query, IDataContext dataContext, Expression expression, object[] ps, object[] preambles, CancellationToken cancellationToken)
LinqToDB.Linq.QueryRunner.ScalarQueryAsync(Query query, IDataContext dataContext, Expression expression, object[] ps, object[] preambles, CancellationToken cancellationToken)
LinqToDB.Linq.QueryRunner+InsertWithIdentity<T>.QueryAsync(IDataContext dataContext, T obj, InsertColumnFilter<T> columnFilter, string tableName, string serverName, string databaseName, string schemaName, TableOptions tableOptions, CancellationToken token)
LinqToDB.DataExtensions.InsertWithInt32IdentityAsync<T>(IDataContext dataContext, T obj, InsertColumnFilter<T> columnFilter, string tableName, string databaseName, string schemaName, string serverName, TableOptions tableOptions, CancellationToken token)
Nop.Data.DataProviders.BaseDataProvider.InsertEntityAsync<TEntity>(TEntity entity) in BaseDataProvider.cs
+
            entity.Id = await dataContext.InsertWithInt32IdentityAsync(entity);
Nop.Data.EntityRepository<TEntity>.InsertAsync(TEntity entity, bool publishEvent) in EntityRepository.cs
+
            await _dataProvider.InsertEntityAsync(entity);
Nop.Services.Customers.CustomerService.InsertGuestCustomerAsync() in CustomerService.cs
+
            await _customerRepository.InsertAsync(customer);
Nop.Web.Framework.WebWorkContext.SetCurrentCustomerAsync(Customer customer) in WebWorkContext.cs
-
                    }
                }
                if (customer == null || customer.Deleted || !customer.Active || customer.RequireReLogin)
                {
                    //create guest if not exists
                    customer = await _customerService.InsertGuestCustomerAsync();
                }
            }
            if (!customer.Deleted && customer.Active && !customer.RequireReLogin)
            {
                //set customer cookie
Nop.Web.Framework.WebWorkContext.GetCurrentCustomerAsync() in WebWorkContext.cs
+
            await SetCurrentCustomerAsync();
Nop.Services.Security.PermissionService.AuthorizeAsync(PermissionRecord permission) in PermissionService.cs
+
            return await AuthorizeAsync(permission, await _workContext.GetCurrentCustomerAsync());
Nop.Web.Framework.Mvc.Filters.CheckAccessPublicStoreAttribute+CheckAccessPublicStoreFilter.CheckAccessPublicStoreAsync(AuthorizationFilterContext context) in CheckAccessPublicStoreAttribute.cs
+
                if (await _permissionService.AuthorizeAsync(StandardPermissionProvider.PublicStoreAllowNavigation))
Nop.Web.Framework.Mvc.Filters.CheckAccessPublicStoreAttribute+CheckAccessPublicStoreFilter.OnAuthorizationAsync(AuthorizationFilterContext context) in CheckAccessPublicStoreAttribute.cs
-
            /// Called early in the filter pipeline to confirm request is authorized
            /// </summary>
            /// <param name="context">Authorization filter context</param>
            /// <returns>A task that represents the asynchronous operation</returns>
            public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
            {
                await CheckAccessPublicStoreAsync(context);
            }
            #endregion
        }
        #endregion
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, object state, bool isCompleted)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
Nop.Services.Authentication.AuthenticationMiddleware.InvokeAsync(HttpContext context) in AuthenticationMiddleware.cs
+
            await _next(context);
StackExchange.Profiling.MiniProfilerMiddleware.Invoke(HttpContext context) in MiniProfilerMiddleware.cs
WebMarkupMin.AspNetCore6.WebMarkupMinMiddlewareBase.Invoke(HttpContext context)
Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
Nop.Services.Installation.InstallUrlMiddleware.InvokeAsync(HttpContext context, IWebHelper webHelper) in InstallUrlMiddleware.cs
+
            await _next(context);
Nop.Services.Common.KeepAliveMiddleware.InvokeAsync(HttpContext context, IWebHelper webHelper) in KeepAliveMiddleware.cs
+
            await _next(context);
Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.HandleException(HttpContext context, ExceptionDispatchInfo edi)
Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

![image](https://user-images.githubusercontent.com/7950917/175104141-620d8ff6-4e16-4ab0-9b90-9b11c9c8e29b.png)
This is the image from exception page to make this section more understandable.

**Comment by sencagri** [comment]
Hi again,
After long hours I MIGHT understand the problem. I think this happened because I swithced to "develop" branch to sync my repo and run the program once. Then I go back to "4.50-bug-fixes" branch then this happened because there are significant difference between 4.5 and 4.6 version especially in the customer table. 

TL;DR I have unwillingly upgraded to 4.60 then swithed back to 4.50

**Comment by exileDev** [comment]
Thanks. Closed #6289
